packages/trader/src/AppV2/Components/TradeParameters/Duration/__tests__/duration_container.spec.tsx

```diff
@@ -0,0 +1,174 @@
+                <DurationActionSheetContainer {...hours} />
+            </TraderProviders>
+        );
+    };
+
+    it('should render the DurationActionSheetContainer with default values', () => {
+        renderDurationContainer(default_trade_store);
+        expect(screen.getByText('Duration')).toBeInTheDocument();
+        expect(screen.getByText('Save')).toBeInTheDocument();
+    });
+
+    it('should select duration in hours if duration is more than 59 minutes', () => {
+        default_trade_store.modules.trade.duration = 130;
+        renderDurationContainer(default_trade_store);
+
+        const duration_chip = screen.getByText('1 h');
+        userEvent.click(duration_chip);
+
+        expect(default_trade_store.modules.trade.onChangeMultiple).not.toHaveBeenCalled();
+    });
+
+    it('should call onChangeMultiple with correct data with hours', async () => {
+        default_trade_store.modules.trade.duration = 130;
+        renderDurationContainer(default_trade_store, { selected_hour: [2, 10], setSelectedHour: jest.fn() });
+
+        userEvent.click(screen.getByText('Save'));
+
+        expect(default_trade_store.modules.trade.onChangeMultiple).toHaveBeenCalledWith({
+            duration_unit: 'm',
+            duration: 130,
+            expiry_time: null,
+            expiry_type: 'duration',
+        });
+    });
+
+    it('should call onChangeMultiple with correct data with ticks', () => {
+        default_trade_store.modules.trade.duration_units_list = ['t'];
+        default_trade_store.modules.trade.duration = 5;
+        default_trade_store.modules.trade.duration_unit = 't';
+
+        renderDurationContainer(default_trade_store);
+
+        userEvent.click(screen.getByText('Save'));
+
+        expect(default_trade_store.modules.trade.onChangeMultiple).toHaveBeenCalledWith({
+            duration_unit: 't',
+            duration: 5,
+            expiry_time: null,
+            expiry_type: 'duration',
+        });
+    });
+
+    it('should call onChangeMultiple with correct data with seconds', () => {
+        default_trade_store.modules.trade.duration = 20;
+        default_trade_store.modules.trade.duration_unit = 's';
+
+        renderDurationContainer(default_trade_store);
+        userEvent.click(screen.getByText('22 sec'));
+        userEvent.click(screen.getByText('Save'));
+
+        expect(default_trade_store.modules.trade.onChangeMultiple).toHaveBeenCalledWith({
+            duration_unit: 's',
+            duration: 20,
+            expiry_time: null,
+            expiry_type: 'duration',
+        });
+    });
+
+    it('should show Expiry Date when days are selected', () => {
+        default_trade_store.modules.trade.duration_unit = 'd';
+        renderDurationContainer(default_trade_store);
+        expect(screen.getByText('Expiry')).toBeInTheDocument();
+    });
+
+    it('should open datepicker on clicking on calendar icon in the days wheelpicker', async () => {
```

Comment:
Why some cases are async and some not? In both you are using userEvent. Sorry if irrelevant